### DISCUSSION TITLE
Scan of the main ticket checks if an add-on product is included.

### DISCUSSION THREAD

**Discussion by Problemkerze** [pretix/pretix#2433] [Ideas] [answered]
Let's say I sell my customers event catering with the booking.
Then it makes sense to make this an add-on product to the main ticket, because one order often contains more than one attendee ticket.

Goal is that participants only need to have one ticket with them.
At the entrance to the catering area, I would therefore like to scan the QR code of the main ticket and check whether the person has booked catering: At a check-in-point, it should be checked whether a main ticket contains a certain add-on product.

Icing on the cake: I have different main tickets that for which catering is bookable and since it's a 3-day event, you can book 1-day catering and 3-day catering (that would be solvable through check-in-lists, I guess).

Is this possible?

**Comment by raphaelm** [comment]
This is currently not possible. We'd very much like to have this feature and we will have it at some point, but we're unfortunately there yet.

In theory, we could extend the check-in process in a way that first checks if the scanned ticket is valid on the selected check-in list. If not, we could look if any of the add-ons of the ticket is valid on the selected check-in list. There are some challenges that make it hard to implement:

* It requires information about the add-on tickets to be available for the check-in decision. This is easy if we're online, possible for our regular offline mode, but almost impossible for our recently introduced offline scanning mode with signed barcodes that does not require offline devices to have the full data set. While this is not a total blocker, we always try to avoid features that "only work in some cases" because it can be hard to understand and surprising for users why they suddenly don't work in other cases.

* A ticket can have multiple add-on tickets. What do we do in the case where *multiple* of the add-ons are valid on the check-in list that is being used for scanning? One option would be showing a selection dialog, but that will be quite an implementation effort and also doesn't work well for non-interactive situations like scanning at a turnstile. So I guess for an initial implementations we'd just need to show an error and block entirely in this case.

And generally, it's a significant new check-in feature that we need to support in both online and offline mode on pretixSCAN for Android, Desktop, and iOS, as well as the web-check-in; that's just a lot of work and takes a while, even after the other ones are solved.

**Reply by Problemkerze** [reply]
Too bad that this won't be a thing. Maybe a viable feature for a future release.

**Reply by raphaelm** [reply]
Absolutely, it will be there in some future release!

**Reply by choseh** [reply]
We have the same use case. Right now we just add the booked catering options as text to the ticket, which works :)

**Comment by ajasja** [comment, marked as answer]
This now seem to be implemented with the `Allow checking in add-on tickets by scanning the main ticket` option in the advanced tab of check-in list
![image](https://github.com/user-attachments/assets/0f91f4af-5df1-4afc-b6b8-2d68a9305a7a)
